### PR TITLE
bumping PL versions & compatibility

### LINKED ISSUE #76: Request to un-pin the PyTorch Lightning Dependency
**Is your feature request related to a problem? Please describe.**

We integrate PyTorch-Tabular in [Lightning-Flash](https://github.com/PyTorchLightning/lightning-flash) thanks to the work of @Actis92

It's awesome!!! Giving us access to a great range of tabular models. However, we've run in to issues where users who install flash for tabular have to have the specific PL version that is pinned by PyTorch-Tabular (e.g. 1.3.6 in the latest release, 1.5.10 on master).

**Describe the solution you'd like**

Would it be an option to just set a minimal version of PL in PyTorch-Tabular but allow for newer? This would make it much more flexible for use as a dependency in other projects.

If there are concerns about compatibility with latest PL versions, I would be happy to help with adding PyTorch-Tabular to the Ecosystem CI (see #63 for details) so that you get nightly GPU tests running with latest PL. Let me know your thoughts :smiley:

**Describe alternatives you've considered**

We will just live with the limitation in Flash 😆

### PR DESCRIPTION
**Pull request by Borda**
let's bump the PL version to the latest and if possible release it...
Fixes #76

### CODE REVIEW

**Comment by Borda** [comment]
this error seems to be unrelated to the PL, as I downgraded PL back to the original and still seeing it... :confused: in `examples/to_test_regression_custom_models.py`
```
omegaconf.errors.ConfigAttributeError: Missing key _model_name
    full_key: _model_name
    object_type=dict
```

**Comment by Borda** [comment]
@manujosephv fyi, I am running all the tests in fork CI, and so far so good, except for the dumping callback, is it really needed to keep/preserve them for any reason? :rabbit:

**Comment by manujosephv** [comment]
@Borda Thank you for taking the time to go through all of these.. I have been slacking off on the repo due to some other commitments. But since the last few weeks, I've been working on an pretty major update to the repo.. major refactoring and addition of functionalities etc. It's almost done and I'll push the new version in a week or so.. I would love it if we can take a look at all this after the new update? If that is too much re-work, we can connect briefly to figure out the best way to make the re-work as less as possible... 

I agree, the examples and some of the tutorials have dead code in them.. and currently re-vamping them as well..

Also, I'll try and bump PL version to the latest one this release.. 

Incidentally, the `_model_name` error I just fixed today :)

**Comment by Borda** [comment]
> Also, I'll try and bump PL version to the latest one this release..

I think that most of the changes here shall be safe, so if possible I would be happier to get this merge before additional refactoring... as I made it during my holiday and sire how much time I would be able to spend later on it... :innocent:

**Comment by Borda** [comment]
BTW, I can try to rever some changes which are not essential, to reduce eventual collisions...

File: docs/index.md [hunk 1]
```diff
@@ -105,13 +106,13 @@ If you use PyTorch Tabular for a scientific publication, we would appreciate cit
...
-    publisher={Zenodo}, 
-    author={manujosephv}, 
-    year={2021}, 
+@article{manujosephv_2021,
+    title={manujosephv/pytorch_tabular: v0.5.0-alpha},
+    DOI={10.5281/zenodo.4732773},
+    abstractNote={<p>First Alpha Release</p>},
+    publisher={Zenodo},
+    author={manujosephv},
+    year={2021},
```
Comment (Borda) [inline docs/index.md:115]:
```suggestion
@article{manujosephv_2021, 
    title={manujosephv/pytorch_tabular: v0.5.0-alpha}, 
    DOI={10.5281/zenodo.4732773}, 
    abstractNote={<p>First Alpha Release</p>}, 
    publisher={Zenodo}, 
    author={manujosephv}, 
    year={2021}, 
```

File: docs/index.md [hunk 2]
```diff
@@ -95,7 +96,7 @@ If you use PyTorch Tabular for a scientific publication, we would appreciate cit
 * [arxiv Paper](https://arxiv.org/abs/2104.13638)
 ```
 @misc{joseph2021pytorch,
-      title={PyTorch Tabular: A Framework for Deep Learning with Tabular Data}, 
+      title={PyTorch Tabular: A Framework for Deep Learning with Tabular Data},
```
Comment (Borda) [inline docs/index.md:99]:
```suggestion
      title={PyTorch Tabular: A Framework for Deep Learning with Tabular Data}, 
```

File: README.md [hunk 3]
```diff
@@ -152,15 +152,15 @@ If you use PyTorch Tabular for a scientific publication, we would appreciate cit
 ```
 * Zenodo Software Citation
 ```
-@article{manujosephv_2021, 
-    title={manujosephv/pytorch_tabular: v0.7.0-alpha}, 
-    DOI={10.5281/zenodo.5359010}, 
+@article{manujosephv_2021,
+    title={manujosephv/pytorch_tabular: v0.7.0-alpha},
+    DOI={10.5281/zenodo.5359010},
```
Comment (Borda) [inline README.md:157]:
```suggestion
@article{manujosephv_2021, 
    title={manujosephv/pytorch_tabular: v0.7.0-alpha}, 
    DOI={10.5281/zenodo.5359010}, 
```

File: README.md [hunk 4]
```diff
@@ -152,15 +152,15 @@ If you use PyTorch Tabular for a scientific publication, we would appreciate cit
...
     abstractNote={<p>Added a few more SOTA models - TabTransformer, FTTransformer
         Made improvements in the model save and load capability
-        Made installation less restrictive by unfreezing some dependencies.</p>}, 
-    publisher={Zenodo}, 
-    author={manujosephv}, 
-    year={2021}, 
+        Made installation less restrictive by unfreezing some dependencies.</p>},
+    publisher={Zenodo},
+    author={manujosephv},
+    year={2021},
```
Comment (Borda) [inline README.md:163]:
```suggestion
        Made installation less restrictive by unfreezing some dependencies.</p>}, 
    publisher={Zenodo}, 
    author={manujosephv}, 
    year={2021}, 
```

File: README.md [hunk 5]
```diff
@@ -142,7 +142,7 @@ If you use PyTorch Tabular for a scientific publication, we would appreciate cit
 * [arxiv Paper](https://arxiv.org/abs/2104.13638)
 ```
 @misc{joseph2021pytorch,
-      title={PyTorch Tabular: A Framework for Deep Learning with Tabular Data}, 
+      title={PyTorch Tabular: A Framework for Deep Learning with Tabular Data},
```
Comment (Borda) [inline README.md:145]:
```suggestion
      title={PyTorch Tabular: A Framework for Deep Learning with Tabular Data}, 
```

**Comment by Borda** [comment]
@manujosephv rebased, could you approve CI run? :otter:

File: .github/workflows/linting.yml [hunk 6]
```diff
@@ -21,4 +21,4 @@ jobs:
         python -m pip install flake8
     - name: Lint with flake8
       run: |
-        flake8 --ignore=W6,E12,W503,E5 pytorch_tabular/ tests/
```
Comment (Borda) [inline .github/workflows/linting.yml:24]: move this to the project-wide config

File: docs/index.md [hunk 7]
```diff
@@ -64,7 +65,6 @@ trainer_config = TrainerConfig(
     auto_lr_find=True, # Runs the LRFinder to automatically derive a learning rate
     batch_size=1024,
     max_epochs=100,
-    gpus=1, #index of the GPU to use. 0, means CPU
```
Comment (Borda) [inline docs/index.md:67]: dropping all these in favor of `accelerator="auto"` which is already in config default :rabbit:

File: docs/tutorials/05-Experiment_Tracking.ipynb [hunk 8]
```diff
@@ -154,7 +154,6 @@
     "    auto_lr_find=True, # Runs the LRFinder to automatically derive a learning rate\n",
     "    batch_size=1024,\n",
     "    max_epochs=100,\n",
-    "    gpus=-1,  #index of the GPU to use. -1 means all available GPUs, None, means CPU\n",
```
Comment (Borda) [inline docs/tutorials/05-Experiment_Tracking.ipynb:157]: dropping all these in favor of `accelerator="auto"` which is already in config default :rabbit:

File: examples/to_test_node.py [hunk 9]
```diff
@@ -91,32 +91,31 @@ def test_classification(
...
-            trainer_config=trainer_config,
-        )
-        tabular_model.fit(train=train, test=test)
-
-        result = tabular_model.evaluate(test)
-        assert result[0]["valid_loss"] < 2.5
-        pred_df = tabular_model.predict(test)
-        assert pred_df.shape[0] == test.shape[0]
+        return
+    data_config = DataConfig(
```
Comment (Borda) [inline examples/to_test_node.py:95]: assert True is the same as just return :)

File: tests/test_common.py [hunk 10]
```diff
@@ -46,28 +41,15 @@
...
 
 def fake_metric(y_hat, y):
     return (y_hat - y).mean()
 
 
-@pytest.mark.parametrize(
-    "model_config_class",
-    MODEL_CONFIG_SAVE_TEST,
-)
-@pytest.mark.parametrize(
```
Comment (Borda) [inline tests/test_common.py:58]: simplify with define columns list only once :chipmunk:

**Review by manujosephv — COMMENTED** [review]
@Borda In the new PyTorch Lightning, `devices=None` automatically picks up all cores/gpus ?

**Pull request merged**